### ISSUE TITLE
flow_from_directory throwing OS Error saying " The filename, directory name, or volume label syntax is incorrect "

### ISSUE THREAD

**Issue by BhShanmukhaBhargav-project** [keras-team/keras#14802]
My code goes this way
`
#IMPORTING LIBRARIES
import tensorflow as tf
from tensorflow.keras.preprocessing.image import ImageDataGenerator
import numpy as np
from tensorflow.keras.preprocessing import  image

#DATA PREPROCESSING

#Preprocessing the Training Set
train_datagen = ImageDataGenerator(rescale = 1./255, shear_range = 0.2, zoom_range = 0.2, horizontal_flip = True)
train_set = train_datagen.flow_from_directory("D:\Code\ML\Datasets\CatsDogs\trainingset", target_size = (64,64), batch_size = 32, class_mode = 'binary')

#Preprocessing the Test Set
test_datagen = ImageDataGenerator(rescale = 1./255)
test_set = test_datagen.flow_from_directory("D:\Code\ML\Datasets\CatsDogs\testset",target_size = (64,64), batch_size = 32, class_mode = 'binary')

#BUILDING THE CNN

#Initialize the CNN
cnn = tf.keras.models.Sequential()
#First Convolution layer
cnn.add(tf.keras.layers.Conv2D(filters = 32, kernel_size = 3, activation = 'relu', input_shape = [64, 64,3]))
#Pooling -->Max Pooling
cnn.add(tf.keras.layers.MaxPool2D(pool_size = 2, strides = 2))
#Second Convolution layer
cnn.add(tf.keras.layers.Conv2D(filters = 32, kernel_size = 3, activation = 'relu'))
#Pooling -->Max Pooling(For the second Convolution layer)
cnn.add(tf.keras.layers.MaxPool2D(pool_size = 2, strides = 2))
#Flattening
cnn.add(tf.keras.layers.Flatten())
#Full Connection
cnn.add(tf.keras.layers.Dense(units = 128, activation = 'relu'))
#output layer
cnn.add(tf.keras.layers.Dense(units = 1, activation = 'sigmoid'))

#TRAINING THE CNN

#Compiling the CNN
cnn.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics = ['accuracy'])
#Training the CNN on train_set and evaluating the same on the test_set at the same time
cnn.fit(x = train_set, validation_data = test_set, epochs = 25)

#MAKING A SINGLE PREDICTION
test_img = image.load_img('D:\Code\ML\Datasets\Cats & Dogs\cat_or_dog_1.jpg', target_size = (64,64))
test_img = image.img_to_array(test_img)
test_img = np.expand_dims(test_img, axis = 0)
result = cnn.predict(test_img)
train_set.class_indices
if result[0][0] == 1:
    prediction = 'dog'
else:
    prediction = 'cat'
print(prediction)

`
I looked into Keras's documentation but couldn't find anything of use.
You can see my directories here.
![image](https://user-images.githubusercontent.com/54494093/123191166-74008d00-d4be-11eb-942e-09140a31cc64.png)
![image](https://user-images.githubusercontent.com/54494093/123191259-9abec380-d4be-11eb-9ac7-dd10c0742a68.png)
Thanks in advance.

**Comment by BhShanmukhaBhargav-project** [comment]
I found the solution for myself  I guess!
I just replaced the file path with forward slashes and it worked out of the box!!
Even though the issue is solved I would like to know why fixing my file path solved the error!!?
